board/fancom/hmi/hmi.c

```diff
@@ -0,0 +1,514 @@
+	 * In some configurations the memory size returned by VideoCore
+	 * is not aligned to the section size, what is mandatory for
+	 * the u-boot's memory setup.
+	 */
+	gd->ram_size &= ~MMU_SECTION_SIZE;
+
+	return 0;
+}
+
+#ifdef CONFIG_OF_BOARD
+int dram_init_banksize(void)
+{
+	int ret;
+
+	ret = fdtdec_setup_memory_banksize();
+	if (ret)
+		return ret;
+
+	return fdtdec_setup_mem_size_base();
+}
+#endif
+
+static void set_fdtfile(void)
+{
+	const char *fdtfile;
+
+	if (env_get("fdtfile"))
+		return;
+
+	fdtfile = model->fdtfile;
+	env_set("fdtfile", fdtfile);
+}
+
+/*
+ * If the firmware provided a valid FDT at boot time, let's expose it in
+ * ${fdt_addr} so it may be passed unmodified to the kernel.
+ */
+static void set_fdt_addr(void)
+{
+	//if (env_get("fdt_addr"))
+	//	return;
+
+	if (fdt_magic(fw_dtb_pointer) != FDT_MAGIC)
+		return;
+
+	env_set_hex("fdt_addr", fw_dtb_pointer);
+}
+
+/*
+ * Prevent relocation from stomping on a firmware provided FDT blob.
+ */
+unsigned long board_get_usable_ram_top(unsigned long total_size)
+{
+	if ((gd->ram_top - fw_dtb_pointer) > SZ_64M)
+		return gd->ram_top;
+	return fw_dtb_pointer & ~0xffff;
+}
+
+static void set_usbethaddr(void)
+{
+	ALLOC_CACHE_ALIGN_BUFFER(struct msg_get_mac_address, msg, 1);
+	int ret;
+
+	if (!model->has_onboard_eth)
+		return;
+
+	if (env_get("usbethaddr"))
+		return;
+
+	BCM2835_MBOX_INIT_HDR(msg);
+	BCM2835_MBOX_INIT_TAG(&msg->get_mac_address, GET_MAC_ADDRESS);
+
+	ret = bcm2835_mbox_call_prop(BCM2835_MBOX_PROP_CHAN, &msg->hdr);
+	if (ret) {
+		printf("bcm2835: Could not query MAC address\n");
+		/* Ignore error; not critical */
+		return;
+	}
+
+	eth_env_set_enetaddr("usbethaddr", msg->get_mac_address.body.resp.mac);
+
+	if (!env_get("ethaddr"))
+		env_set("ethaddr", env_get("usbethaddr"));
+
+	return;
+}
+
+#ifdef CONFIG_ENV_VARS_UBOOT_RUNTIME_CONFIG
+static void set_board_info(void)
+{
+	char s[11];
+
+	snprintf(s, sizeof(s), "0x%X", revision);
+	env_set("board_revision", s);
+	snprintf(s, sizeof(s), "%d", rev_scheme);
+	env_set("board_rev_scheme", s);
+	/* Can't rename this to board_rev_type since it's an ABI for scripts */
+	snprintf(s, sizeof(s), "0x%X", rev_type);
+	env_set("board_rev", s);
+	env_set("board_name", model->name);
+}
+#endif /* CONFIG_ENV_VARS_UBOOT_RUNTIME_CONFIG */
+
+static void set_serial_number(void)
+{
+	ALLOC_CACHE_ALIGN_BUFFER(struct msg_get_board_serial, msg, 1);
+	int ret;
+	char serial_string[17] = { 0 };
+
+	if (env_get("serial#"))
+		return;
+
+	BCM2835_MBOX_INIT_HDR(msg);
+	BCM2835_MBOX_INIT_TAG_NO_REQ(&msg->get_board_serial, GET_BOARD_SERIAL);
+
+	ret = bcm2835_mbox_call_prop(BCM2835_MBOX_PROP_CHAN, &msg->hdr);
+	if (ret) {
+		printf("bcm2835: Could not query board serial\n");
+		/* Ignore error; not critical */
+		return;
+	}
+
+	snprintf(serial_string, sizeof(serial_string), "%016llx",
+		 msg->get_board_serial.body.resp.serial);
+	env_set("serial#", serial_string);
+}
+
+int misc_init_r(void)
+{
+	set_fdt_addr();
+	//set_fdtfile();
```

Comment:
Why has this been commented out?

Reply:
This was part of a test to get blue/green working. I will reenable and test it.